core/src/main/kotlin/xtdb/util/HashUtil.kt

```diff
@@ -0,0 +1,13 @@
+package xtdb.util
+
+import java.nio.ByteBuffer
+
+fun ByteBuffer.toByteArray(): ByteArray {
+    return if (hasArray() && arrayOffset() == 0 && array().size == remaining()) {
+        array()
+    } else {
+        ByteArray(remaining()).also { get(it) }
+    }
+}
+
+fun byteBufferToBytes(buf: ByteBuffer): ByteArray = buf.toByteArray()
```

Comment:
this can probably go as a default impl on the Hasher interface?

i.e. `fun hash(value: ByteBuffer) = ...`